lib/datasets.go

```diff
@@ -53,22 +53,29 @@ func (m DatasetMethods) Name() string {
 // Attributes defines attributes for each method
 func (m DatasetMethods) Attributes() map[string]AttributeSet {
 	return map[string]AttributeSet{
-		"changereport": {AEChanges, "POST"},
```

Comment:
So, others may disagree, but I don't think we should be changing the Attributes values for GET/POST, even though we're switching (possibly temporarily) to all POST requests all the time. The reason is both aspiration (where we want to get to, if we add more RESTful sugar), and also acts as documentation for whether the method is read-only or read-write. It'll be easier to adapt code in the future if we don't need to undo this modification.

Rather, the function `dispatchMethodCall` in `lib/dispatch.go` is already ignoring this value, and always calling `inst.http.CallMethod` with the "POST" verb. So we've already switched to a POST-only world, and these attributes reflect who the methods *could* act.

Reply:
I'm actually personally against adding the notion of anything except "POST" w/ application/json to anything at the lib level or lower. I think it will be cleaner if lib only understands one way of doing thing, with `api`'s new role to be conforming to any special cases. 

This is slightly off topic, but related. In order to use the callable/regMethods to programmatically allow the instance to add handlers to a server, I'm pitching that actually what we want for AttributeSet is really:

```
type AttributeSet struct {
   endpoint APIEndpoint
   denyRPC bool
}
```

Especially if we find that there are lib methods we are okay with folks accessing via the api, but not rpc. The only one I can think of right now would be something like `ProfilePhotoSet`, which we would need for any frontend work, but we may not want to have over rpc. I'm basing this off the fact that we don't want any config changes over rpc, and profile changes are technically also config changes. However, w/ multi tenancy this is probably no longer the case!

If there IS a one-to-one mapping of non-api endpoints & methods we don't want over rpc, then we can just proceed w/ only having the `APIEndpoint` in the `AttributeSet` & drop the http method type entirely.

Reply:
In this case I would side with the `POST` only option. While I understand that this might expand later and would technically be doubling the work, I'd like to remove as much cruft as possible and get as close as possible to the spec for a clean slate moving forward. Mostly to establish nice patterns which we can build on top rather than bending them here and there.

Unsure whether we should spend any time right now on handling the case where we support API but not RPC on a global level. But even if we did want to do that now, feels like a config thing that should be a switch in the dispatch not on the attribute/method level.